apps/web/app/modules/Courses/Lesson/Lesson.page.tsx

```diff
@@ -85,11 +103,19 @@ export default function LessonPage() {
         <Button
           className="w-min self-end"
           variant="outline"
-          onClick={
-            data?.isSubmitted
-              ? () => clearQuizProgress.mutate({ lessonId })
-              : () => submitQuiz.mutate({ lessonId })
-          }
+          onClick={() => {
+            switch (data?.isSubmitted) {
+              case true:
+                clearQuizProgress.mutate({ lessonId });
+                break;
+              case false:
+                submitQuiz.mutate({ lessonId });
+                setIsOpen(true);
+                break;
+              default:
+                break;
```

Comment:
it could be extracted to function (maybe with ts-pattern?) - what do you think? 🤔